src/components/forms/shipment/ShipmentForm.tsx

```diff
@@ -0,0 +1,190 @@
+    );
+    setOrderDetails(arrayDeleted);
+  };
+
+  const createMultipleShipments = async (
+    payload: FieldValues,
+  ) => {
+    await refreshAccessToken();
+    await postRequest(shipmentUrl, payload);
+  };
+
+  const updateShipment = async (
+    payload: FieldValues,
+  ) => {
+    await refreshAccessToken();
+    await putRequest(
+      shipmentUrl,
+      payload,
+      payload.id,
+    );
+  };
+
+  const updateShipmentOrders = async (
+    id: number,
+    orders: any,
+  ) => {
+    const orderPayload = { orders };
+    if (orderPayload.orders.length > 0) {
+      await refreshAccessToken();
+      await postRequest(
+        `/shipment_details_orders/${id}/add_orders_to_shipment/`,
+        orderPayload,
+      );
+    }
+  };
+
+  const formatShipmentPayload = (formValues: FieldValues, orders: string []) => {
+    const payload = { ...formValues };
+    payload.driver = formValues.driver.id;
+    payload.transporter_details = formValues.transporter_details.id;
+    payload.supplier = formValues.supplier.id;
+    payload.delivery_status = formValues.delivery_status.value;
+    payload.pod_status = formValues.pod_status.value;
+    if (mode === 'Add') {
+      payload.orders = orders;
+    } else {
+      payload.id = initialFormValue.id;
+      delete payload.order_ids;
+      delete payload.order_names;
+      delete payload.orders;
+    }
+    return payload;
+  };
+
+  const onSubmitAdd = async (formValues: FieldValues) => {
+    const customers = Array.from(new Set(orderDetails.map((row: any) => (row.customer))));
+    if (customers.length > 1) {
+      customers.forEach(async (customerId, index) => {
+        const orders = orderDetails
+          .filter((order: any) => order.customer === customerId)
+          .map((row: any) => (row.order_details_id));
+        const payload = formatShipmentPayload(formValues, orders);
+        await createMultipleShipments(payload);
+        if (index === customers.length - 1) {
+          navigate(`${Paths.shipment_details}`);
+        }
+      });
+    } else {
+      const orders = orderDetails.map((row: any) => (row.order_details_id));
+      const payload = formatShipmentPayload(formValues, orders);
+      dispatch(postShipmentRequest(payload as PostShipmentRequestPayload));
+      navigate(`${Paths.shipment_details}`);
+    }
+  };
+
+  const onSubmitEdit = async (formValues: FieldValues) => {
+    const orders = orderDetails.map((row: any) => (row.order_details_id));
+    const payload = formatShipmentPayload(formValues, orders);
+    await updateShipment(payload);
+    await updateShipmentOrders(payload.id, orders);
+    navigate(`${Paths.shipment_details}`);
+  };
+
+  const onSubmit = async (formValues: FieldValues) => {
+    if (mode === 'Add') {
+      onSubmitAdd(formValues);
+    } else {
+      onSubmitEdit(formValues);
+    }
+  };
+
+  return (
+    <div className="bg-transit-white w-full rounded-lg pt-8 px-4" style={{ width: '75%', margin: '0 auto' }}>
```

Comment:
First you added width 100% and then width 75% in style props - maybe just remove style props and write `w-3/4 my-auto`

Reply:
this will be fixed later.